projects/flow.aggregation-mpi/project.py

```diff
@@ -0,0 +1,42 @@
+from flow import FlowProject, aggregator, directives
+
+
+class Project(FlowProject):
+    pass
+
+
+RANKS_PER_JOB = 2
+JOBS_PER_AGGREGATE = 8
+
+
+@Project.operation
+@directives(nranks=lambda *jobs: RANKS_PER_JOB * len(jobs))
+@aggregator.groupsof(JOBS_PER_AGGREGATE)
+def do_mpi_task(*jobs):
+    from mpi4py import MPI
+
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+    size = comm.Get_size()
+    num_jobs = len(jobs)
+    print(f"Launching MPI tasks for {len(jobs)} jobs. Rank {rank} of {size}.")
+
+    # Use MPI splitting to make new communicators
+    communicator_size = size // num_jobs
+    color = rank // communicator_size
+    key = rank % communicator_size
+    print(f"{rank=}, {color=}, {key=}")
+
+    split_comm = comm.Split(color, key)
+    split_rank = split_comm.Get_rank()
+    split_size = split_comm.Get_size()
+    job = jobs[color]
```

Comment:
```suggestion

    # Select the job from the aggregate to run in the split communicator.
    job = jobs[color]
```
This is the key step, it needs a little more explanation.